octobot_trading/data/trade.py

```diff
@@ -14,23 +14,54 @@
 #  You should have received a copy of the GNU Lesser General Public
 #  License along with this library.
 
-from octobot_trading.data.order import Order
+import time
+
+from octobot_trading.enums import OrderStatus
 
 
 class Trade:
-    def __init__(self, order: Order):
-        self.order = order
-        self.currency, self.market = self.order.get_currency_and_market()
-        self.quantity = self.order.filled_quantity
-        self.price = self.order.filled_price
-        self.cost = self.order.total_cost
-        self.order_type = self.order.order_type
-        self.final_status = self.order.status
-        self.fee = self.order.fee
-        self.order_id = self.order.order_id
-        self.side = self.order.side
-        self.creation_time = self.order.creation_time
-        self.canceled_time = self.order.canceled_time
-        self.filled_time = self.order.executed_time
-        self.symbol = self.order.symbol
-        self.simulated = self.order.trader.simulate
+    def __init__(self, trader):
+        self.trader = trader
+        self.exchange_manager = trader.exchange_manager
+
+        self.status = OrderStatus.OPEN
+        self.creation_time = time.time()
+        self.executed_time = 0
+
+        self.trade_id = trader.parse_order_id(None)
+
+        self.symbol = None
+        self.currency = None
+        self.market = None
+        self.taker_or_maker = None
+        self.timestamp = 0
+        self.origin_price = 0
+        self.created_last_price = 0
+        self.origin_quantity = 0
+        self.origin_stop_price = 0
+        self.trade_type = None
+        self.side = None
+        self.executed_quantity = 0
+        self.canceled_time = 0
+        self.executed_time = 0
+        self.fee = None
+        self.executed_price = 0
+        self.trade_profitability = 0
+        self.total_cost = 0
+
+    def update_from_order(self, order, creation_time=0, canceled_time=0, executed_time=0):
+        self.currency, self.market = order.get_currency_and_market()
+        self.executed_quantity = order.filled_quantity
+        self.executed_price = order.filled_price
+        self.origin_price = order.origin_price
+        self.origin_quantity = order.origin_quantity
+        self.total_cost = order.total_cost
+        self.trade_type = order.order_type
+        self.status = order.status
+        self.fee = order.fee
+        self.trade_id = order.order_id
+        self.side = order.side
+        self.creation_time = order.creation_time if order.creation_time > 0 else creation_time
+        self.canceled_time = order.canceled_time if order.canceled_time > 0 else canceled_time
+        self.executed_time = order.executed_time if order.executed_time > 0 else executed_time
+        self.symbol = order.symbol
```

Comment:
It looks like ```taker_or_maker```, ```created_last_price``` and ```origin_stop_price``` are not updated. Not sure if ```created_last_price``` and ```origin_stop_price``` are really relevant but ```taker_or_maker``` should be updated

Reply:
Right !